### PR TITLE
add calico networking doc

### PR DESCRIPTION
**Pull request by mainred**
## High-level description

This PR introduces the newly added services,  system requirements, marathon application examples, etc. This will be used also as a reference to introduce calico in dcos-site.

**For the newly added components, we should add calico components in the diagram shown in https://docs.d2iq.com/mesosphere/dcos/1.13/overview/architecture/components in the following doc PR.**

## Corresponding DC/OS tickets (obligatory)

These DC/OS JIRA ticket(s) must be updated (ideally closed) in the moment this PR lands:

  - [DCOS-58586](https://jira.mesosphere.com/browse/DCOS-58586) document for Calico in DC/OS

## Checklist for all PRs

  - [ ] Added a comprehensible changelog entry to `CHANGES.md` or explain why this is not a user-facing change:
  - [ ] Included a test which will fail if code is reverted but test is not. If there is no test please explain here:
  - [ ] Read the [DC/OS contributing guidelines](https://github.com/dcos/dcos/blob/master/contributing.md)
  - [ ] Followed relevant code rules [Rules for Packages and Systemd](https://github.com/dcos/dcos/tree/master/docs)

## Checklist for component/package updates:

If you are changing components or packages in DC/OS (e.g. you are bumping the sha or ref of anything underneath `packages`), then in addition to the above please also include:

  - [ ] Change log from the last version integrated (this should be a link to commits for easy verification and review): [example](https://github.com/dcos/dcos-mesos-modules/compare/f6fa27d7c40f4207ba3bb2274e2cfe79b62a395a...6660b90fbbf69a15ef46d0184e36755881d6a5ae)
  - [ ] Test Results: [link to CI job test results for component]
  - [ ] Code Coverage (if available): [link to code coverage report]
___
**PLEASE FILL IN THE TEMPLATE ABOVE** / **DO NOT REMOVE ANY SECTIONS ABOVE THIS LINE**

## Instructions and review process

**What is the review process and when will my changes land?**

All PRs require 2 approvals using GitHub's [pull request reviews](https://help.github.com/articles/about-pull-request-reviews/).

Reviewers should be:
* Developers who understand the code being modified.
* Developers responsible for code that interacts with or depends on the code being modified.

It is best to proactively ask for 2 reviews by @mentioning the candidate reviewers in the PR comments area. The responsibility is on the developer submitting the PR to follow-up with reviewers and make sure a PR is reviewed in a timely manner. Once a PR has **2 ship-it's**, **no red reviews**, and **all tests are green** it will be included in the [next train](https://github.com/dcos/dcos/blob/master/contributing.md).

### CODE REVIEW

File: packages/calico/README.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+## Configuration Reference(Networking)
+
+| Parameter            | Description                                                                                                                        |
+|----------------------|------------------------------------------------------------------------------------------------------------------------------------|
+| calico_enabled       | Indicates whether to enable Calico networking.[Default: 'true']                                                                                           |
+| calico_network_cidr  | Subnet allocated for calico. The subnet specified by `calico_network_cidr` should not overlay with those for VXLAN backends or virtual networks defined for [DC/OS virtual networks](https://github.com/mesosphere/dcos-docs-site/blob/staging/pages/mesosphere/dcos/1.14/installing/production/advanced-configuration/configuration-reference/index.md#dcos_overlay_enable). Default: '192.168.0.0/16']                                                                          |
```
Comment (vespian) [inline packages/calico/README.md]:
```suggestion
| calico_network_cidr  | Subnet allocated for calico. The subnet specified by `calico_network_cidr` should not overlap with those for VXLAN backends or virtual networks defined for [DC/OS virtual networks](https://github.com/mesosphere/dcos-docs-site/blob/staging/pages/mesosphere/dcos/1.14/installing/production/advanced-configuration/configuration-reference/index.md#dcos_overlay_enable). Default: '192.168.0.0/16']                                                                          |
```

File: packages/calico/README.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+| calico_vxlan_port    | The UDP port used for calico VXLAN. This configuration works when calico_vxlan_enabled is set to be true. [Default: 4789]           |
+| calico_vxlan_vni     | The virtual network ID used for calico VXLAN. This configuration works when calico_vxlan_enabled is set to be true. [Default: 4096] |
+| calico_vxlan_mtu     | The MTU to set on the VXLAN tunnel device. This configuration works when calico_vxlan_enabled is set to be true. Please refer to [this](https://docs.projectcalico.org/v3.8/networking/mtu) for a suitable MTU configuration [Default: 1410]    |
+| calico_felix_prometheusmetrics_port     | TCP port that the Prometheus metrics server should bind to. [Default: 9091] |
+
+
+## Example
+
+To use Calico networking containers, you only have to specify the network name as `calico`, and the following examples show two Marathon application definition of Calico networking containers supported by Mesos UCR and Docker runtime respectively.
```
Comment (vespian) [inline packages/calico/README.md]:
```suggestion
To use Calico networking containers, you only have to specify the network name as `dcos`, and the following examples show two Marathon application definition of Calico networking containers supported by Mesos UCR and Docker runtime respectively.
```

Comment (mainred) [reply packages/calico/README.md]: The network name should be `calico`, `dcos` is used to create a dcos-overlay network container

File: packages/calico/README.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+## Configuration Reference(Networking)
+
+| Parameter            | Description                                                                                                                        |
+|----------------------|------------------------------------------------------------------------------------------------------------------------------------|
+| calico_enabled       | Indicates whether to enable Calico networking.[Default: 'true']                                                                                           |
+| calico_network_cidr  | Subnet allocated for calico. The subnet specified by `calico_network_cidr` should not overlap with those for VXLAN backends or virtual networks defined for [DC/OS virtual networks](https://github.com/mesosphere/dcos-docs-site/blob/staging/pages/mesosphere/dcos/1.14/installing/production/advanced-configuration/configuration-reference/index.md#dcos_overlay_enable). Default: '192.168.0.0/16']                                                                          |
```
Comment (mainred) [inline packages/calico/README.md]: The default value of `calico_network_cidr` should be empty, and we should also add a description that `this value should not conflict with host network`.

File: CHANGES.md [hunk 4]
```diff
@@ -34,6 +34,10 @@ Please follow the [`CHANGES.md` modification guidelines](https://github.com/dcos
 
 * Add etcd into DC/OS. (DCOS-59004)
 
+* Calico in DC/OS: introduced Calico network for mesos UCR, enabled by `calico_enabled` with default value `true`. (DCOS-58582)
```
Comment (mainred) [inline CHANGES.md]: TODO: we will replace these items with just one like `Introducing calico in DC/OS`

**Pull request closed without merging**